lib/stateTransition/validation/validateStateTransitionFeeFactory.js

```diff
@@ -0,0 +1,53 @@
+const ValidationResult = require('../../validation/ValidationResult');
+const stateTransitionTypes = require('../../stateTransition/stateTransitionTypes');
+const InvalidStateTransitionTypeError = require('../../errors/InvalidStateTransitionTypeError');
+const IdentityBalanceIsNotEnoughError = require('../../errors/IdentityBalanceIsNotEnoughError');
+
+const PRICE_PER_BYTE = 1;
+
+/**
+ * Validate state transition fee
+ *
+ * @param {DataProvider} dataProvider
+ * @return {validateStateTransitionFee}
+ */
+function validateStateTransitionFeeFactory(dataProvider) {
+  /**
+   * @typedef validateStateTransitionFee
+   * @param {DataContractStateTransition|DocumentsStateTransition} stateTransition
+   * @return {ValidationResult}
+   */
+  async function validateStateTransitionFee(stateTransition) {
+    const result = new ValidationResult();
+
+    const serializedStateTransition = stateTransition.serialize({ skipSignature: true });
+    const byteSize = Buffer.byteLength(serializedStateTransition);
+    const feeSize = byteSize * PRICE_PER_BYTE;
+
+    let identityId;
+    switch (stateTransition.getType()) {
+      case stateTransitionTypes.DATA_CONTRACT:
+        identityId = stateTransition.getDataContract().getOwnerId();
+        break;
+      case stateTransitionTypes.DOCUMENTS:
+        identityId = stateTransition.getDocuments()[0].getOwnerId();
+        break;
+      default:
```

Comment:
We need to handle IdentityCreateTransition too, but in this case, we need to validate against lock transaction output

Reply:
@shumkov Why don't we validate the fee in this method as well? A data provider is set, so it's possible to fetch the transaction. The validity of the outpoint can be verified in the validateData. What's the exact reason you want to move it from here?

Reply:
We call `validateFee` on `checkTx` but I'm not sure that we want to go to Dash Core RPC.

Reply:
Maybe we shouldn't call `validateFee` for `IdentityCreateTransition` on `checkTx`?